### PR TITLE
[char] about.md

### PR DESCRIPTION
**Pull request by bobahop**
I realize this will be tricky to get right.

### CODE REVIEW

File: concepts/char/about.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
 # About
 
-TODO
+Rust implements the [char][char type] type to represent a single character. A `char` literal is placed within single quotes, like `'a'`.
+Each `char` is four bytes in size, which is large enough to represent most [Unicode][unicode] characters as [Unicode Scalar Values][unicode scalar].
```
Comment (coriolinus) [inline concepts/char/about.md:4]:
```suggestion
Each `char` is four bytes in size, which is large enough to represent [Unicode][unicode] characters as [Unicode Scalar Values][unicode scalar].
```

I'm not aware of any unicode codepoints which can't be represented as `char`.

File: concepts/char/about.md [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+
+However, a [char][character type] is not always what we think of as a letter. There are some languages, e.g. Hindi,  that use [diacritics][diacritics],
```
Comment (coriolinus) [inline concepts/char/about.md:6]:
```suggestion
However, a [`char`][character type] is not always what we think of as a letter. There are some languages, e.g. Hindi,  that use [diacritics][diacritics],
```

File: concepts/char/about.md (hunk 1 above)
Comment (bobahop) [reply concepts/char/about.md:4]: "The char type represents a single character. More specifically, since 'character' isn't a well-defined concept in Unicode, char is a 'Unicode scalar value', which is similar to, but not the same as, a 'Unicode code point'." 

"Unicode Scalar Value. Any Unicode code point _except high-surrogate and low-surrogate code points_. In other words, the ranges of integers 0 to D7FF16 and E00016 to 10FFFF16 inclusive."

https://doc.rust-lang.org/std/primitive.char.html
http://www.unicode.org/glossary/#unicode_scalar_value

File: concepts/char/about.md [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+which are special symbols which modify the character they are attached to. Although the diacritic in Rust is a separate `char`, it is the diacritic and
+the character it modifies that we commonly think of as a letter.
+
+The term for a character and its diacritic is [`grapheme cluster`][grapheme cluster]. There are external crates that can be used to process grapheme clusters,
```
[the remark this answers is not in the document]
Comment (bobahop) [reply concepts/char/about.md:10]: I'm not sure I understand how you would want it changed.

[the remark this answers is not in the document]
Comment (coriolinus) [reply concepts/char/about.md:10]: I think that he's suggesting that we can omit the backticks which cause the phrase to render as code.

[the remark this answers is not in the document]
Comment (efx) [reply concepts/char/about.md:10]: Exactly @coriolinus ! Sorry about the confusion @bobahop.

[the remark this answers is not in the document]
Comment (bobahop) [reply concepts/char/about.md:10]: Oh, yes, you're absolutely right, plus the backticks hide the link.

File: concepts/char/about.md [hunk 4]
```diff
@@ -1,3 +1,42 @@
...
 
-TODO
+Rust implements the [char][char type] type to represent a single character. A `char` literal is placed within single quotes, like `'a'`.
+Each `char` is four bytes in size, which is large enough to represent most [Unicode][unicode] characters as [Unicode Scalar Values][unicode scalar].
+
+However, a [`char`][character type] is not always what we think of as a letter. There are some languages, e.g. Hindi,  that use [diacritics][diacritics],
+which are special symbols which modify the character they are attached to. Although the diacritic in Rust is a separate `char`, it is the diacritic and
+the character it modifies that we commonly think of as a letter.
+
+The term for a character and its diacritic is [`grapheme cluster`][grapheme cluster]. There are external crates that can be used to process grapheme clusters,
```
[the remark this answers is not in the document]
Comment (bobahop) [reply concepts/char/about.md]: Sorry, I just committed the backticks removal independently.

File: concepts/char/about.md (hunk 1 above)
Comment (coriolinus) [reply concepts/char/about.md:4]:
```suggestion
Each `char` is four bytes in size and represents a single [Unicode Scalar Values][unicode scalar].
```

I think this version is both well-defined and simpler; if unicode characters aren't a well-defined concept, let's just omit mentioning them.

**Review by coriolinus — APPROVED** [review]
Thanks for putting in the work to write this!

File: concepts/char/about.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (efx) [reply concepts/char/about.md]: @bobahop no need to apologize! I do not care about suggestion attribution. I realized it would be clearer for me to make the suggestion than explain in plain English.
